test/index.js

```diff
@@ -513,7 +513,9 @@ test('equals', function() {
   eq(Z.equals(Math.sin, Math.cos), false);
   eq(Z.equals(Identity(Identity(Identity(0))), Identity(Identity(Identity(0)))), true);
   eq(Z.equals(Identity(Identity(Identity(0))), Identity(Identity(Identity(1)))), false);
-  eq(Z.equals({'@@type': 'my-package/Quux'}, {'@@type': 'my-package/Quux'}), false);
+  eq(Z.equals({'@@type': 'my-package/Quux'}, {'@@type': 'my-package/Quux'}), true);
```

Comment:
That's right. Note how `Z.equals` behaves on `master` when giving a prototype object:

```javascript
> var Z = require('.')
undefined
> var Maybe = require('./test/Maybe')
undefined
> Maybe.prototype
Maybe {
  '@@type': 'sanctuary-type-classes/Maybe',
  'fantasy-land/equals': [Function],
  'fantasy-land/map': [Function],
  'fantasy-land/alt': [Function],
  toString: [Function],
  inspect: [Function] }
> Z.equals(Maybe.prototype, Maybe.prototype)
undefined
```

How does `Z.equals(Maybe.prototype, Maybe.prototype)` evaluate to `undefined`?! By dispatching to `Maybe.prototype['fantasy-land/equals']` **with `Maybe.prototype` as the implicit argument**. Here's the method's definition:

```javascript
Maybe.prototype[FL.equals] = function(other) {
  return this.isNothing ? other.isNothing
                        : other.isJust && Z.equals(other.value, this.value);
};
```

We essentially perform this method invocation:

```javascript
Maybe.prototype['fantasy-land/equals'](Maybe.prototype)
```

Substituting `Maybe.prototype` for `this` and `other`:

```javascript
Maybe.prototype.isNothing ?
  Maybe.prototype.isNothing :
  Maybe.prototype.isJust && Z.equals(Maybe.prototype.value, Maybe.prototype.value)
```

Evaluating the condition:

```javascript
undefined ?
  Maybe.prototype.isNothing :
  Maybe.prototype.isJust && Z.equals(Maybe.prototype.value, Maybe.prototype.value)
```

Taking the appropriate branch:

```javascript
Maybe.prototype.isJust && Z.equals(Maybe.prototype.value, Maybe.prototype.value)
```

Evaluating the left-hand operand:

```javascript
undefined && Z.equals(Maybe.prototype.value, Maybe.prototype.value)
```

Short circuiting:

```javascript
undefined
```

This makes it clear that methods should not receive a prototype object as `this`.